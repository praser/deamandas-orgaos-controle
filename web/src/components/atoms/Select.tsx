import React from 'react';
import SelectSearch from 'react-select-search';
import SelectProps from './SelectProps';
import './Select.css';

const Select = (props: SelectProps): JSX.Element => (
  <span data-testid="select">
    <SelectSearch {...props} />;
  </span>
);

export default Select;
