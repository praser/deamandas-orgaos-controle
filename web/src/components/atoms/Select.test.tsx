import React from 'react';
import { render, screen } from '@testing-library/react';
import Select from './Select';

describe('Select atomic component', () => {
  let select: HTMLElement;
  const testId = 'select';

  beforeEach(() => {
    const data = [
      { value: 'a', name: 'a' },
      { value: 'b', name: 'b' },
      { value: 'c', name: 'c' },
      { value: 'd', name: 'd' },
    ];

    render(<Select options={data} />);
    select = screen.getByTestId(testId);
  });

  it('is expected to match snapshot', () => {
    expect(select).toMatchSnapshot();
  });
});
