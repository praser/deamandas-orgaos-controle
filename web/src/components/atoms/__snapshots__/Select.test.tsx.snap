// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select atomic component is expected to match snapshot 1`] = `
<span
  data-testid="select"
>
  <div
    class="select-search"
  >
    <div
      class="select-search__value"
    >
      <input
        autocomplete="on"
        class="select-search__input"
        readonly=""
        tabindex="0"
        value="a"
      />
    </div>
  </div>
  ;
</span>
`;
